examples/ola-recorder.cpp

```diff
@@ -115,36 +122,70 @@ int VerifyShow(const string &filename) {
     return ola::EXIT_NOINPUT;
 
   map<unsigned int, unsigned int> frames_by_universe;
-  uint64_t total_time = 0;
 
-  unsigned int universe;
-  ola::DmxBuffer buffer;
-  unsigned int timeout;
+  ShowEntry entry;
   ShowLoader::State state;
+  uint64_t playback_pos = 0;
+  bool playing = false;
   while (true) {
-    state = loader.NextFrame(&universe, &buffer);
+    state = loader.NextEntry(&entry);
     if (state != ShowLoader::OK)
       break;
-    frames_by_universe[universe]++;
-
-    state = loader.NextTimeout(&timeout);
-    if (state != ShowLoader::OK)
+    playback_pos += entry.next_wait;
+    frames_by_universe[entry.universe]++;
```

Comment:
Wouldn't this make the clamping routine redundant, and is probably clearer to read? Subject to the `playback_pos > FLAGS_start` bit?

```suggestion
    if (playing)
      frames_by_universe[entry.universe]++;
    } else {
      // Clamp the frame count to 1 as we haven't actually started playing yet
      frames_by_universe[entry.universe] = 1;
    }
```

Reply:
That's a really great way of simplifying this!